client/src/document/ingredients/browser-compatibility-table/index.tsx

```diff
@@ -130,6 +163,11 @@ export default function BrowserCompatibilityTable({
   locale: string;
 }) {
   const location = useLocation();
+  const gleanClick = useGleanClick();
+
+  const observedNode = useViewed(() => {
```

Comment:
This will not work for large BCD tables. `useViewed` only works if 50% of the component fit on the screen. It was made for placements and it also works with the scrims. As it's crucial for us I would suggest we use `useIsIntersecting` directly, the 1s timeout is not really important for this.

Reply:
I looked into using `useIsIntersecting()`, but it requires significant boilerplate here.

How about observing the header instead? See 1d36f8d6c0167f7a64337c1fbbaab21987d48e11.

Reply:
I pushed https://github.com/mdn/yari/pull/12030/commits/16a1f2653adb8161790a8ea55e5b6a0bf678b9a0 to allow configuring useViewed

I'm fine with using "just" the header, your choice whether you want to decide on proper options and use the whole table or if the header is good enough.

Reply:
Thanks, indeed `threshold: 0` resolves the issue with long tables, see: 143719fb70b4f23fe87bb4bdb6f3c8da1bc38e20